kore/src/SMT/SimpleSMT.hs

```diff
@@ -263,9 +263,20 @@ send solver@Solver { hIn } command' = do
 
 recv :: Solver -> IO SExpr
 recv solver@Solver { hOut } = do
-    resp <- Text.hGetLine hOut
+    responseLines <- readResponse 0 []
+    let resp = Text.intercalate "\n" (reverse responseLines)
     info solver ["recv"] resp
     readSExpr resp
+  where
+    readResponse :: Int -> [Text] -> IO [Text]
```

Comment:
Could you add some comment describing what this is doing? I'm not sure I understand.

Reply:
Done.